### ISSUE TITLE
Presence of TS_CAL1 on STM32L011

### ISSUE THREAD

**Issue by badcf00d** [STMicroelectronics/STM32CubeL0#29]
The LL API seems to incorrectly assume that TS_CAL1 (temperature sensor calibration value @ 30c) is missing on the STM32L011 series.

The datasheet says in section 6.3.16 Temperature Sensor Characteristics, sub-point 2, "V30 ADC conversion result is stored in the TS_CAL1 byte".

I've tested this on STM32L011 development boards, and the TS_CAL1 value is indeed present in 0x1FF8007A, and the value seems sensible, producing quite accurate temperature readings using the included macro.

Could stm32l0xx_ll_adc.h be modified to correct this? Thanks.

**Comment by RJMSTM** [comment]
Hello @badcf00d 

Would you please give us more details about how you got this issue? 
Could you please share a screenshot of the value read in addresses 0x1FF8007A (TS_CAL1)  or share the project you have used to reproduce the issue in order to allow a better analysis of the problem.

With regards,
Rania

**Comment by badcf00d** [comment]
Hi @RJMSTM

The issue is just that there are multiple comments and #defines in `stm32l0xx_ll_adc.h` that say that TS_CAL1 isn't available on STM32L011:

https://github.com/STMicroelectronics/STM32CubeL0/blob/bec4e499a74de98ab60784bf2ef1912bee9c1a22/Drivers/STM32L0xx_HAL_Driver/Inc/stm32l0xx_ll_adc.h#L196-L198 and
https://github.com/STMicroelectronics/STM32CubeL0/blob/bec4e499a74de98ab60784bf2ef1912bee9c1a22/Drivers/STM32L0xx_HAL_Driver/Inc/stm32l0xx_ll_adc.h#L1461-L1467

But this seems to be incorrect. `*((uint16_t*) 0x1FF8007AU)` on one of my dev boards is `649`, another it's `664`, and they both give accurate temperature readings when I remove the #defines and use the `__LL_ADC_CALC_TEMPERATURE` macro.

So I think the fix is just to remove the comments about TS_CAL1 being unavailable on L011, and remove `STM32L011xx` from the #defines that I mentioned above.

Thanks,

**Comment by badcf00d** [comment]
Hi @RJMSTM,

Thanks for getting back to me, like I say it seems like the TS_CAL1 value is present on the L011, all the dev boards I've got give accurate temperature within a couple of degrees when using it. So yeh maybe it's just been left off the datasheet by accident or something.

Regards,

**Comment by RJMSTM** [comment]
Hello @badcf00d 

Please allow me to close this thread as no activity. You may reopen it at any time if you have any details to share with us in order to help you to solve the issue. Thank you for your comprehension.

With regards

**Comment by badcf00d** [comment]
I don't think this is ready to be closed? This still seems to be a bug in the LL API:

https://github.com/STMicroelectronics/STM32CubeL0/blob/e416b1d1ede2719bc253eb8c8029e8e338567f2f/Drivers/STM32L0xx_HAL_Driver/Inc/stm32l0xx_ll_adc.h#L195C81-L195C81

If you would like any more information let me know, but basically, `0x1FF8007AU` seems to hold TS_CAL1 on L011, but the LL API is wrong and says TS_CAL1 is not present on L011.

**Comment by ALABSTM** [comment]
Hi @badcf00d,

I hear your point. As shown on the screenshot below, the TS_CAL**1** mention you are referring to in section 6.3.16 of the datasheet is in a footnote and is very likely to be a typo. The main table, plus the revision history (section 9), both refer to TS_CAL**2**. TS_CAL**1** is officially not supported on STM32L011xx devices.

If you have a look at the same 6.3.16 section in the datasheet of another device like the [STM32L031xx](https://www.st.com/resource/en/datasheet/stm32l031c4.pdf), for instance, you'll see that the table lists both calibration values.

With these compilation switches, the driver is consistent with the official documentation. Please allow me to close this issue. Thank you for your understanding and thank you again for having tackled this point.

With regards,

**PS:** Regarding the typo in the datasheet, feel free to report it to the [ST Community](https://community.st.com/t5/stm32-mcus-products/bd-p/stm32-mcu-products-forum).

<img width="1431" height="792" alt="Image" src="https://github.com/user-attachments/assets/560b9752-0d1d-462b-9b4e-16d2a157cf92" />
